src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/remote.py

```diff
@@ -197,6 +197,15 @@ def get_remote_os_version(self):
         if self._remote_os_version is None:
             try:
                 self._remote_os_version = self.client.call('system.version')
+            except AttributeError:
```

Comment:
Why its an AttributeError?

Reply:
`self.client` gets set to `None` when the remote client disconnects.

Reply:
Shoudnt we check for that then first? Instead of catching exception

Reply:
`self._remote_os_version` gets set to None as well on remote client disconnect but `self.client` gets set to None first so I could move the check for that first.

Reply:
err, actually no `self._remote_os_version` gets set to `None` first. That's why I'm catching the exception @william-gr

Reply:
I still don't get it, shouldnt matter, as long as check for self.client